src/miqclient/filters.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from .utils import escape_filter
+
+OPERATORS = {u'=', u'!=', u'<', u'<=', u'>=', u'>'}
+
+
+def gen_filter(name, op, value, is_or=False):
+    """Generates a single filter expression for ``filter[]``."""
+    if op not in OPERATORS:
+        raise ValueError('Unknown operator {}'.format(op))
+    result = u'{} {} {}'.format(name, op, escape_filter(value))
+    if is_or:
+        result = u'or ' + result
+    return result
+
+
+class Q(object):
+    """A Django-like query composition class.
+
+    Note:
+        Only supports simple chaining using ``|`` and ``&`` (as ``filter[]`` supports)
+
+    Args:
+        name: Name of the value to be compared
+        op: An operator
+        value: Value to be matched.
+
+    .. code-block:: python
+
+        api.collections.vms.filter(Q('name', '=', 'foo') | Q('name', '=', 'bar'))
+
+    Et cetera ... You can use ``|`` and ``&`` operators and they only work in chain (no parentheses)
+    because that is what filter supports.
+
+    """
+    @classmethod
+    def from_dict(cls, d):
+        """Creates a query (AND and =) from a dictionary."""
```

Comment:
Are you assuming AND between every key, value pair in the dictionary? If so, why not allowing OR?